solutions/rust/03-ping-pong-multiple/diff/src/main.rs.diff

```diff
@@ -0,0 +1,34 @@
+@@ -1,23 +1,30 @@
+ use std::io::{Read, Write};
+ use std::net::TcpListener;
+
+ fn main() {
+     let listener = TcpListener::bind("127.0.0.1:6379").unwrap();
+
+     for stream in listener.incoming() {
+         match stream {
+             Ok(mut stream) => {
+                 println!("accepted new connection");
+
+-                // Wait for the client to send us a message but ignore the content for now
+                 let mut buf = [0; 512];
+-                stream.read(&mut buf).unwrap();
+
+-                stream.write("+PONG\r\n".as_bytes()).unwrap();
++                loop {
++                    // Wait for the client to send us a message but ignore the content for now
++                    let bytes_read = stream.read(&mut buf).unwrap();
```

Comment:
@ThetaSinner when `buf` is passed to `stream.read` like this repeatedly, does it override the buffer contents on every read? Or does it append to the buffer?

If it's the latter, we might want to move the `buf` assignment into the loop. I know it doesn't matter here since we aren't reading from it, just making sure that this doesn't seem like a bug to users.

Reply:
@ThetaSinner Ah, got it. Just to confirm - if you read "hello" into the buffer the first time, and "hey" the next time, will the buffer contain `hey`? or will it contain `heylo`? (i.e. some bytes were overwritten but the others weren't)